tests/test-sessions/test-file-sessions/src/test/java/org/eclipse/jetty/server/session/FileTestHelper.java

```diff
@@ -177,14 +177,9 @@ public static void createFile (String id, String contextPath, String vhost,
 
             if (attributes != null)
             {
-                List<String> keys = new ArrayList<String>(attributes.keySet());
-                out.writeInt(keys.size());
+                SessionData tmp = new SessionData(id,contextPath, vhost, created, accessed, lastAccessed, maxIdle);
                 ObjectOutputStream oos = new ObjectOutputStream(out);
-                for (String name:keys)
-                {
-                    oos.writeUTF(name);
-                    oos.writeObject(attributes.get(name));
-                }
+                SessionData.serializeAttributes(tmp, oos);
```

Comment:
`ObjectOutputStream` not closed, use try-with-resources.

Reply:
The ObjectOutputStream is wrapping a FileOutputStream - it's inappropriate to close the OOS because it will close the FOS.

Reply:
Ok, so `DataOutputStream out` can be included in the try-with-resources at the beginning? Then `ObjectOutputStream oos` can be left as is.

Reply:
Yes, it is.